pkg/cmd/registry/artifacts/crud/get/get.go

```diff
@@ -0,0 +1,164 @@
+package get
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	flagutil "github.com/redhat-developer/app-services-cli/pkg/cmdutil/flags"
+	"github.com/redhat-developer/app-services-cli/pkg/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/iostreams"
+	"github.com/redhat-developer/app-services-cli/pkg/localize"
+	"github.com/redhat-developer/app-services-cli/pkg/serviceregistry/registryinstanceerror"
+	"github.com/spf13/cobra"
+
+	"github.com/redhat-developer/app-services-cli/internal/config"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/factory"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/registry/artifacts/util"
+
+	"github.com/redhat-developer/app-services-cli/pkg/logging"
+)
+
+type Options struct {
+	artifact   string
+	group      string
+	outputFile string
+
+	registryID string
+	version    string
+
+	IO         *iostreams.IOStreams
+	Config     config.IConfig
+	Logger     func() (logging.Logger, error)
+	Connection factory.ConnectionFunc
+	localizer  localize.Localizer
+}
+
+func NewGetCommand(f *factory.Factory) *cobra.Command {
+	opts := &Options{
+		Config:     f.Config,
+		Connection: f.Connection,
+		IO:         f.IOStreams,
+		localizer:  f.Localizer,
+		Logger:     f.Logger,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "get",
+		Short: "Get artifact by id and group",
+		Long: `Get artifact by specifying id and group.
+Command will fetch the latest artifact from the registry based on the artifactId and group.
+
+When --version is specified command will fetch the specific version of the artifact.
+Get command will fetch artifacts based on group and artifactId and version.
+For fetching artifacts using global identifiers please use "service-registry download" command
+`,
+		Example: `
+## Get latest artifact by name
+rhoas service-registry artifacts get myschema
+
+## Get latest artifact and save its content to file
+rhoas service-registry artifacts get myschema myschema.json
+
+## Get latest artifact and pipe it to other command 
+rhoas service-registry artifacts get myschema | grep -i 'user'
+
+## Get latest artifact by specifying custom group, registry and name as flag
+rhoas service-registry artifacts get --group mygroup --registryId=myregistry --artifact myartifact
+`,
+		Args: cobra.RangeArgs(0, 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.artifact = args[0]
+			}
+
+			if opts.artifact == "" {
+				return fmt.Errorf("Artifact is required. Please specify artifact as positional argument or by using --artifact flag")
```

Comment:
It is probably better to use one or the other (positional or flag). Offering both does not give much functional benefit but adds more documentation.

Reply:
That is intentional for UX trial.
I do not like our positional flags usage in our current cli. They poorly documented and not isomorphic (some commands take name or id. Some just id and if we mix them we getting 404 with some missleading message.

For registry it is hard to define positionals as there are multiple different Ids, groups etc.

So my take is to see how users will use those commands and then adjust/remove positionals and flags. Can we keep duplicates for now (before adding i18n )?

Reply:
> They poorly documented and loosely defined (some commands take name or id.
> Positionals have very clear usage in other CLIs but in our CLI they seem little bit overused.

All names using a positional argument. Eg: `rhoas kafka describe [name]`
All IDs use a flag. Eg: `rhoas kafka describe --id [id]`.

I would not say that they are **over**used. They are used very little.

> For registry it is hard to define positionals that would be easy to remember for user as there are multiple different Ids, groups etc. and each command will have different mixture of those.

You should really only use it for a `name` argument, if there is any.

> So my take is to see how users will use those commands and then adjust/remove positionals and flags.

How do you plan to collect usage infrmation?

Reply:
We will send guide to users to try CLI and then have them sent us usage log.
More info:
https://github.com/aerogear/charmil/issues/222